public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx

```diff
@@ -58,19 +58,48 @@ interface DispatchProps {
   updateTimeZoneForSession: typeof updateTimeZoneForSession;
 }
 
+interface State {
+  topPanelSize: string | number;
+  rightPanelSize: string | number;
+}
+
 type Props = OwnProps & ConnectedProps & DispatchProps;
 
-export class PanelEditorUnconnected extends PureComponent<Props> {
+export class PanelEditorUnconnected extends PureComponent<Props, State> {
   querySubscription: Unsubscribable;
+  rafToken: RefObject<number>;
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      topPanelSize: props.uiState.topPaneSize,
+      rightPanelSize: props.uiState.rightPaneSize,
+    };
+
+    this.rafToken = createRef<number>();
+  }
 
   componentDidMount() {
     this.props.initPanelEditor(this.props.sourcePanel, this.props.dashboard);
+
+    window.addEventListener('resize', this.updateSplitPaneSize);
   }
 
   componentWillUnmount() {
     this.props.panelEditorCleanUp();
+    window.removeEventListener('resize', this.updateSplitPaneSize);
   }
 
+  updateSplitPaneSize = () => {
+    if (this.rafToken.current !== undefined) {
+      window.cancelAnimationFrame(this.rafToken.current!);
+    }
+    window.requestAnimationFrame(() => {
+      this.setState({ rightPanelSize: this.props.uiState.rightPaneSize });
+      this.setState({ topPanelSize: this.props.uiState.topPaneSize });
```

Comment:
Can we do in a single pass?
```suggestion
      this.setState({ 
        rightPanelSize: this.props.uiState.rightPaneSize,
        topPanelSize: this.props.uiState.topPaneSize
      });
```

Reply:
Gah, yes! No idea what happened there :smile: